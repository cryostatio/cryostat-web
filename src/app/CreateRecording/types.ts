/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { EventTemplate, KeyValue } from '@app/Shared/Services/api.types';
import { ValidatedOptions } from '@patternfly/react-core';

export type EventTemplateIdentifier = Pick<EventTemplate, 'name' | 'type'>;

interface _FormBaseData {
  name: string;
  template?: EventTemplateIdentifier;
  labels: KeyValue[];
  continuous: boolean;
  archiveOnStop: boolean;
  restart: boolean;
  duration: number;
  durationUnit: number;
  skipDurationCheck?: boolean;
  maxAge: number;
  maxAgeUnit: number;
  maxSize: number;
  maxSizeUnit: number;
  toDisk: boolean;
}

interface _FormValidationData {
  nameValid: ValidatedOptions;
  labelsValid: ValidatedOptions;
  durationValid: ValidatedOptions;
}

export type CustomRecordingFormData = _FormBaseData & _FormValidationData;
