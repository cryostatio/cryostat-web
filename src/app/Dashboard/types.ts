/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FeatureLevel } from '@app/Shared/Services/service.types';
import { gridSpans, LabelProps } from '@patternfly/react-core';
import { Observable } from 'rxjs';

export interface Sized<T> {
  minimum: T;
  default: T;
  maximum: T;
}

export interface DashboardCardSizes {
  span: Sized<gridSpans>;
  height: Sized<number>;
}

export interface DashboardCardDescriptor {
  featureLevel: FeatureLevel;
  icon?: React.ReactNode;
  labels?: {
    content: string;
    color?: LabelProps['color'];
    icon?: React.ReactNode;
  }[];
  preview?: React.ReactNode;
  title: string;
  cardSizes: DashboardCardSizes;
  description: string;
  descriptionFull: JSX.Element | string;
  component: DashboardCardFC<DashboardCardTypeProps>;
  propControls: PropControl[];
  advancedConfig?: JSX.Element;
}
export type DashboardCardFC<P> = React.FC<P> & {
  cardComponentName: string;
};

export interface PropControlExtra {
  displayMapper?: (value: string) => string /* only has effect with 'select' PropControl kind */;
  min?: number;
  max?: number;
  [key: string]: unknown;
}

export interface PropControl {
  name: string;
  key: string;
  description: string;
  kind: 'boolean' | 'number' | 'string' | 'text' | 'select';
  values?: unknown[] | Observable<unknown>;
  defaultValue: unknown;
  extras?: PropControlExtra;
}

export interface DashboardCardTypeProps {
  span: number;
  dashboardId: number;
  isDraggable?: boolean;
  isResizable?: boolean;
  isFullHeight?: boolean;
  actions?: JSX.Element[];
}

export interface ValidationError {
  message: React.ReactNode;
}

export interface CardValidationResult {
  errors: ValidationError[];
  callForAction?: React.ReactNode;
}

export type LayoutTemplateFilter = 'Suggested' | 'Cryostat' | 'User-submitted';

export interface LayoutTemplateController {
  selectedTemplate: SelectedLayoutTemplate | undefined;
  setSelectedTemplate: (template: React.SetStateAction<SelectedLayoutTemplate | undefined>) => void;
  isUploadModalOpen: boolean;
  setIsUploadModalOpen: (isOpen: React.SetStateAction<boolean>) => void;
}

export interface CardConfig {
  id: string;
  name: string;
  span: gridSpans;
  props: object;
}

export type SerialCardConfig = Omit<CardConfig, 'id'>;

export const mockSerialCardConfig: SerialCardConfig = {
  name: 'Default',
  span: 12,
  props: {},
};
export interface DashboardLayout {
  name: string;
  cards: CardConfig[];
  favorite: boolean;
}

export type SerialDashboardLayout = Omit<DashboardLayout, 'name' | 'cards'> & { cards: SerialCardConfig[] };

// only name and vendor are needed to identify a template
export type LayoutTemplateRecord = Pick<LayoutTemplate, 'name' | 'vendor'>;

export enum LayoutTemplateVersion {
  'v2.3' = 'v2.3',
  'v2.4' = 'v2.4',
}

export enum LayoutTemplateVendor {
  BLANK = 'Blank',
  CRYOSTAT = 'Cryostat',
  USER = 'User-submitted',
}

export interface LayoutTemplate {
  name: string;
  description: string;
  cards: SerialCardConfig[];
  version: LayoutTemplateVersion;
  vendor: LayoutTemplateVendor;
}

export interface SelectedLayoutTemplate {
  template: LayoutTemplate;
  category: LayoutTemplateFilter;
}

export type SerialLayoutTemplate = Omit<LayoutTemplate, 'vendor'>;
