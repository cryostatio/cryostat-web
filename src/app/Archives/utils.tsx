/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { RecordingDirectory, Target } from '@app/Shared/Services/api.types';

export const includesDirectory = (arr: RecordingDirectory[], dir: RecordingDirectory): boolean => {
  return arr.some((t) => t.connectUrl === dir.connectUrl);
};

export const indexOfDirectory = (arr: RecordingDirectory[], dir: RecordingDirectory): number => {
  let index = -1;
  arr.forEach((d, idx) => {
    if (d.connectUrl === dir.connectUrl) {
      index = idx;
    }
  });
  return index;
};

export const getTargetFromDirectory = (dir: RecordingDirectory): Target => {
  return {
    connectUrl: dir.connectUrl,
    alias: dir.jvmId,
    labels: [],
    annotations: {
      cryostat: [],
      platform: [],
    },
  };
};
