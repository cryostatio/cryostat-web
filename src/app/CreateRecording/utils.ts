/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { TemplateType } from '@app/Shared/Services/api.types';
import { EventTemplateIdentifier } from './types';

export const RecordingNamePattern = /^[\w_]+$/;

export const DurationPattern = /^[1-9][0-9]*$/;

export const isRecordingNameValid = (name: string) => RecordingNamePattern.test(name);

export const isDurationValid = (duration: number) => DurationPattern.test(`${duration}`);

export const templateFromEventSpecifier = (eventSpecifier: string): EventTemplateIdentifier | undefined => {
  const regex = /^template=([a-zA-Z0-9]+)(?:,type=([a-zA-Z0-9]+))?$/im;
  if (eventSpecifier && regex.test(eventSpecifier)) {
    const parts = regex.exec(eventSpecifier);
    if (parts) {
      return {
        name: parts[1],
        type: parts[2] as TemplateType,
      };
    }
  }
  return undefined;
};
