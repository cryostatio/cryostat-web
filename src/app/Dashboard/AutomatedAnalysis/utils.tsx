/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AnalysisResult, Evaluation } from '@app/Shared/Services/api.types';
import { Stack, StackItem, Label, Title, Text } from '@patternfly/react-core';
import _ from 'lodash';

export const transformAADescription = (result: AnalysisResult): JSX.Element => {
  const format = (s: Evaluation): JSX.Element => {
    if (typeof s === 'string') {
      return <Text>{s}</Text>;
    }
    if (Array.isArray(s)) {
      return (
        <Stack>
          {s.map((e) => (
            <StackItem key={e.setting}>
              <Title headingLevel={'h6'}>{e.setting}</Title>
              <Label>
                {e.name}={e.value}
              </Label>
            </StackItem>
          ))}
        </Stack>
      );
    }
    throw `Unrecognized item: ${s}`;
  };
  return (
    <div>
      {Object.entries(result.evaluation || {}).map(([k, v]) =>
        v && v.length ? (
          <div key={k}>
            <span>
              <Title headingLevel={'h5'}>{_.capitalize(k)}</Title>
              {format(result.evaluation[k])}
            </span>
            <br />
          </div>
        ) : (
          <div key={k}></div>
        ),
      )}
    </div>
  );
};
