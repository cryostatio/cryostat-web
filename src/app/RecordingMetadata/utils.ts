/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { KeyValue } from '@app/Shared/Services/api.types';
import { ValidatedOptions } from '@patternfly/react-core';
import { Observable, from } from 'rxjs';

export const isEqualLabel = (a: KeyValue, b: KeyValue) => {
  return a.key === b.key && a.value === b.value;
};

export const includesLabel = (arr: KeyValue[], searchLabel: KeyValue) => {
  return arr.some((l) => isEqualLabel(searchLabel, l));
};

export const parseLabelsFromFile = (file: File): Observable<KeyValue[]> => {
  return from(
    file
      .text()
      .then(JSON.parse)
      .then((obj) => {
        const labels: KeyValue[] = [];
        const labelObj = obj['labels'];
        if (labelObj) {
          Object.keys(labelObj).forEach((key) => {
            labels.push({
              key: key,
              value: labelObj[key],
            });
          });
          return labels;
        }
        throw new Error('No labels found in file');
      }),
  );
};

export const getLabelDisplay = (label: KeyValue) => `${label.key}:${label.value}`;

export const LabelPattern = /^\S+$/;

export const getValidatedOption = (isValid: boolean) => {
  return isValid ? ValidatedOptions.success : ValidatedOptions.error;
};

export const matchesLabelSyntax = (l: KeyValue) => {
  return l && LabelPattern.test(l.key) && LabelPattern.test(l.value);
};
