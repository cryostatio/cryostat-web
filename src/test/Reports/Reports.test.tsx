/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Reports } from '@app/Reports/Reports';
import { Target } from '@app/Shared/Services/api.types';
import { defaultServices } from '@app/Shared/Services/Services';
import '@testing-library/jest-dom';
import { of } from 'rxjs';
import { renderSnapshot } from '../utils';

const mockFooTarget: Target = {
  id: 1,
  jvmId: 'abcd1234',
  agent: false,
  connectUrl: 'service:jmx:rmi://someFooUrl',
  alias: 'fooTarget',
  labels: [],
  annotations: {
    cryostat: [],
    platform: [],
  },
};

jest.spyOn(defaultServices.targets, 'targets').mockReturnValue(of([mockFooTarget]));

jest.spyOn(defaultServices.notificationChannel, 'messages').mockReturnValue(of());

jest.spyOn(defaultServices.api, 'getCurrentReportsForAllTargets').mockReturnValue(
  of([
    {
      target: mockFooTarget,
      hasSources: true,
      report: {
        lastUpdated: 1752088003576,
        aggregate: {
          count: 1,
          max: 75,
        },
        data: [
          {
            key: 'a',
            value: {
              name: 'a',
              topic: 'topic1',
              score: 75,
              evaluation: {
                summary: 'a summary',
                explanation: 'a explanation',
                solution: 'a solution',
                suggestions: [],
              },
            },
          },
        ],
      },
    },
  ]),
);

jest.mock('@app/Recordings/TargetAnalysis', () => {
  return {
    AutomatedAnalysisResults: jest.fn((_) => {
      return <div>Automated Analysis Results</div>;
    }),
  };
});

jest.mock('@app/Topology/Entity/EntityDetails', () => {
  return {
    EntityDetails: jest.fn((_) => {
      return <div>Entity Details</div>;
    }),
  };
});

describe('<Reports />', () => {
  it('renders correctly', async () => {
    const tree = await renderSnapshot({
      routerConfigs: {
        routes: [
          {
            path: '/recordings',
            element: <Reports />,
          },
        ],
      },
    });
    expect(tree?.toJSON()).toMatchSnapshot();
  });
});
