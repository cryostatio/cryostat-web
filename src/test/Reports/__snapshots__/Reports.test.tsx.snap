// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reports /> renders correctly 1`] = `
<div
  className="pf-v5-c-page__main-group"
>
  <section
    className="pf-v5-c-page__main-section pf-m-fill"
  >
    <nav
      aria-label="Breadcrumb"
      className="pf-v5-c-breadcrumb"
      data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
      data-ouia-component-type="PF5/Breadcrumb"
      data-ouia-safe={true}
    >
      <ol
        className="pf-v5-c-breadcrumb__list"
        role="list"
      >
        <li
          className="pf-v5-c-breadcrumb__item"
        >
          <h1
            className="pf-v5-c-breadcrumb__heading"
          >
            Reports
          </h1>
        </li>
      </ol>
    </nav>
    <div
      className="pf-v5-l-stack pf-m-gutter"
    >
      <div
        className="pf-v5-l-stack__item"
      >
        <div
          className="pf-v5-c-card pf-m-compact"
          data-ouia-component-id="OUIA-Generated-Card-1"
          data-ouia-component-type="PF5/Card"
          data-ouia-safe={true}
          id=""
        >
          <div
            className="pf-v5-c-card__body"
          >
            <div
              className="pf-v5-l-stack pf-m-gutter"
            >
              <div
                className="pf-v5-l-stack__item"
              >
                <div
                  className="pf-v5-l-split pf-m-gutter"
                >
                  <div
                    className="pf-v5-l-split__item"
                  >
                    <button
                      aria-disabled={false}
                      aria-label={null}
                      className="pf-v5-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-1"
                      data-ouia-component-type="PF5/Button"
                      data-ouia-safe={true}
                      disabled={false}
                      onClick={[Function]}
                      role={null}
                      type="button"
                    >
                      Show All
                    </button>
                  </div>
                  <div
                    className="pf-v5-l-split__item"
                  >
                    <div
                      className="pf-v5-c-toggle-group"
                      role="group"
                    >
                      <div
                        className="pf-v5-c-toggle-group__item"
                      >
                        <button
                          aria-pressed={true}
                          className="pf-v5-c-toggle-group__button pf-m-selected"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            className="pf-v5-c-toggle-group__text"
                          >
                            All Available
                          </span>
                        </button>
                      </div>
                      <div
                        className="pf-v5-c-toggle-group__item"
                      >
                        <button
                          aria-pressed={false}
                          className="pf-v5-c-toggle-group__button"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            className="pf-v5-c-toggle-group__text"
                          >
                            Warning Only
                          </span>
                        </button>
                      </div>
                      <div
                        className="pf-v5-c-toggle-group__item"
                      >
                        <button
                          aria-pressed={false}
                          className="pf-v5-c-toggle-group__button"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            className="pf-v5-c-toggle-group__text"
                          >
                            Critical Only
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    className="pf-v5-l-split__item pf-m-fill"
                  />
                  <div
                    className="pf-v5-l-split__item"
                  >
                    <button
                      aria-disabled={false}
                      aria-label={null}
                      className="pf-v5-c-button pf-m-plain"
                      data-ouia-component-id="OUIA-Generated-Button-plain-1"
                      data-ouia-component-type="PF5/Button"
                      data-ouia-safe={true}
                      disabled={false}
                      onClick={[Function]}
                      role={null}
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        aria-labelledby={null}
                        className="pf-v5-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 1024 1024"
                        width="1em"
                      >
                        <path
                          d="M1016,640.1 L724,640.1 L695.9,608.7 C701.8,597.3 706.7,585.2 710.5,572.7 L793.5,567.2 C799.9,566.3 804.9,559.9 804.9,553.5 L804.9,468.2 C804.8,461.7 800.2,456.2 793.8,454.4 L709.3,449.7 C705.5,437.7 700.5,426.1 694.7,415.1 L750.4,351.5 C754.3,346.3 753.3,338.3 748.8,333.8 L688.5,273.5 C684,269 676.8,268.3 671,271.6 L607.3,328.6 C596.4,322.9 585,318.2 573.2,314.5 L567.6,229.6 C566.7,223.2 560.3,218.2 553.9,218.2 L468.6,218.2 C462.2,218.2 456.7,222.8 454.9,229.2 L450.2,314.2 C438,317.9 426.3,322.8 415.2,328.6 L351.1,273.6 C345.9,269.7 337.9,270.7 333.4,275.2 L304.3,304.3 L240.6,240.6 C310.1,171.1 406.1,128 512,128 C718.3,128 887.2,291.6 895.6,495.9 C896,504.5 903.1,511.2 911.6,511.2 L1007.7,511.2 C1016.7,511.2 1024,503.7 1023.7,494.7 C1014.6,220.4 788.6,0 512.1,0 C370.9,0 242.9,57.5 150.2,150.1 L13.7,13.6 C8.6,8.6 0,12.2 0,19.3 L0,376.1 C0,380.5 3.6,384.1 8,384.1 L299.9,384.1 L328.1,414.6 C322,426 316.9,438 312.9,450.5 L230.1,456.9 C223.7,457.8 218.7,464.2 218.7,470.6 L218.7,555.9 C218.7,562.3 223.3,567.8 229.7,569.6 L312,572.9 C315.9,585.9 321,598.3 327.2,610.1 L274,672.1 C270.1,677.3 271.1,685.3 275.6,689.8 L335.9,750.1 C340.4,754.6 347.6,755.3 353.4,752 L413.3,696.7 C425.2,703.1 437.9,708.3 451,712.3 L457.2,793.1 C458.1,799.5 464.5,804.5 470.9,804.5 L556.2,804.5 C562.6,804.5 568.1,799.9 569.9,793.5 L573.1,711.7 C585.9,707.7 598.2,702.5 609.9,696.2 L671.4,750.1 C676.6,754 684.6,753 689.1,748.5 L718.9,718.7 L783.6,783.4 C714,852.9 618,896 512.1,896 C305.4,896 136.4,732.1 128.4,527.3 C128,518.7 121,512 112.4,512 L16.3,512 C7.3,512 0,519.5 0.3,528.5 C9,803.2 235.3,1024 512.1,1024 C653.2,1024 781.2,966.6 873.9,873.9 L1010.3,1010.4 C1015.4,1015.4 1024,1011.8 1024,1004.7 L1024,648 C1024,643.6 1020.4,640 1016,640.1 M511.1,640.9 C440.4,640.9 383.1,583.6 383.1,512.9 C383.1,442.2 440.4,384.9 511.1,384.9 C581.8,384.9 639.1,442.2 639.1,512.9 C639.1,583.6 581.8,640.9 511.1,640.9"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="pf-v5-l-stack__item"
              >
                <div
                  className="pf-v5-c-content"
                >
                  Automated Analysis Reports which have been previously generated and are present in Cryostat's cache will appear here. Reports can be generated 
                  <a
                    href="/recordings#report"
                    onClick={[Function]}
                  >
                    manually on-demand
                  </a>
                   or 
                  <a
                    href="/rules/create"
                    onClick={[Function]}
                  >
                    automatically on a recurring basis
                  </a>
                  .
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pf-v5-l-stack__item"
      >
        <div
          className="pf-v5-c-pagination"
          data-ouia-component-id="OUIA-Generated-Pagination-top-1"
          data-ouia-component-type="PF5/Pagination"
          data-ouia-safe={true}
          id="options-menu-top-pagination"
        >
          <div
            className="pf-v5-c-pagination__total-items"
          >
            <b>
              1
               - 
              1
            </b>
             
            of
             
            <b>
              1
            </b>
             
            
          </div>
          <button
            aria-expanded={false}
            aria-haspopup="listbox"
            className="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
            data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-1"
            data-ouia-component-type="PF5/MenuToggle"
            data-ouia-safe={true}
            disabled={false}
            id="options-menu-top-toggle"
            onClick={[Function]}
            type="button"
          >
            <span
              className="pf-v5-c-menu-toggle__text"
            >
              <b>
                1
                 - 
                1
              </b>
               
              of
               
              <b>
                1
              </b>
               
              
            </span>
            <span
              className="pf-v5-c-menu-toggle__controls"
            >
              <span
                className="pf-v5-c-menu-toggle__toggle-icon"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 320 512"
                  width="1em"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  />
                </svg>
              </span>
            </span>
          </button>
          <nav
            aria-label="Pagination"
            className="pf-v5-c-pagination__nav"
          >
            <div
              className="pf-v5-c-pagination__nav-control pf-m-first"
            >
              <button
                aria-disabled={true}
                aria-label="Go to first page"
                className="pf-v5-c-button pf-m-plain pf-m-disabled"
                data-action="first"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF5/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 448 512"
                  width="1em"
                >
                  <path
                    d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                  />
                </svg>
              </button>
            </div>
            <div
              className="pf-v5-c-pagination__nav-control"
            >
              <button
                aria-disabled={true}
                aria-label="Go to previous page"
                className="pf-v5-c-button pf-m-plain pf-m-disabled"
                data-action="previous"
                data-ouia-component-id="OUIA-Generated-Button-plain-3"
                data-ouia-component-type="PF5/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                  />
                </svg>
              </button>
            </div>
            <div
              className="pf-v5-c-pagination__nav-page-select"
            >
              <span
                className="pf-v5-c-form-control pf-m-disabled"
              >
                <input
                  aria-invalid={false}
                  aria-label="Current page"
                  data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                  data-ouia-component-type="PF5/TextInput"
                  data-ouia-safe={true}
                  disabled={true}
                  max={1}
                  min={1}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  required={false}
                  type="number"
                  value={1}
                />
              </span>
              <span
                aria-hidden="true"
              >
                of
                 
                1
              </span>
            </div>
            <div
              className="pf-v5-c-pagination__nav-control"
            >
              <button
                aria-disabled={true}
                aria-label="Go to next page"
                className="pf-v5-c-button pf-m-plain pf-m-disabled"
                data-action="next"
                data-ouia-component-id="OUIA-Generated-Button-plain-4"
                data-ouia-component-type="PF5/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </button>
            </div>
            <div
              className="pf-v5-c-pagination__nav-control pf-m-last"
            >
              <button
                aria-disabled={true}
                aria-label="Go to last page"
                className="pf-v5-c-button pf-m-plain pf-m-disabled"
                data-action="last"
                data-ouia-component-id="OUIA-Generated-Button-plain-5"
                data-ouia-component-type="PF5/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 448 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                  />
                </svg>
              </button>
            </div>
          </nav>
          <div
            className="pf-v5-l-stack pf-m-gutter"
          >
            <div
              className="pf-v5-l-stack__item"
            >
              <div
                className="pf-v5-c-card pf-m-compact"
                data-ouia-component-id="OUIA-Generated-Card-2"
                data-ouia-component-type="PF5/Card"
                data-ouia-safe={true}
                id=""
              >
                <div
                  className="pf-v5-c-card__body"
                >
                  <div
                    className="pf-v5-l-split pf-m-gutter"
                  >
                    <div
                      className="pf-v5-l-split__item"
                      style={
                        Object {
                          "minHeight": "40em",
                          "width": "35%",
                        }
                      }
                    >
                      <div>
                        <p>
                          Entity Details
                        </p>
                        <div>
                          fooTarget
                        </div>
                      </div>
                    </div>
                    <div
                      className="pf-v5-l-split__item pf-m-fill"
                    >
                      <div>
                        <p>
                          Automated Analysis Results
                        </p>
                        <div>
                          1752088003576
                        </div>
                        <div>
                          {"name":"a","topic":"topic1","score":75,"evaluation":{"summary":"a summary","explanation":"a explanation","solution":"a solution","suggestions":[]}}
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
